mlflow/utils/search_utils.py

```diff
@@ -2011,6 +2011,141 @@ def _get_comparison(cls, comparison):
         return comp
 
 
+@dataclass
+class TraceMetricsFilter:
+    view_type: str
+    entity: str
+    key: str | None
+    comparator: str
+    value: Any
+
+
+class SearchTraceMetricsUtils(SearchTraceUtils):
+    # Entities that are supported for each view type
+    # Mapping of entity to a boolean indicating if it requires a key
+    # For example, "tag" requires a key: "trace.tag.<key> = <value>"
+    # "status" does not require a key: "trace.status = <value>"
+    _VALID_TRACE_ENTITIES = {
+        "status": False,
```

Comment:
Can we define an enum for these searchable keys of traces such as status or tag?